control/control.go

```diff
@@ -35,51 +32,42 @@ func NewHTTPServer(cfg *Config) (*HTTPServer, error) {
 	}
 
 	mux := http.NewServeMux()
-	addr := net.UnixAddr{
-		Name: cfg.SocketPath,
-		Net: SocketType,
-	}
 
 	return &HTTPServer{
+		Addr: cfg.SocketPath,
 		mux: mux,
-		addr: addr,
-		listening: false,
 	}, nil
 }
 
-var listener net.Listener
-
 // Serve starts serving the control server
-func (s *HTTPServer) Serve() {
-	s.lock.Lock()
-	defer s.lock.Unlock()
-
-	// ref https://github.com/joyent/containerpilot/pull/165
-	if listener != nil {
-		return
-	}
+func (s *HTTPServer) Start(app interface{}) {
+	// s.lock.Lock()
+	// defer s.lock.Unlock()
```

Comment:
The reason we had a lock around serving the endpoint was that if we reloaded the configuration multiple times, we could get into a race situation where the first reload worked and then the second reload would kill the application because of the previously-bound socket. Have we avoided that situation now?